src/app/services/extend/extend.service.ts

```diff
@@ -18,33 +18,68 @@
  */
 
 import { Injectable } from '@angular/core';
-import { ExtendDatabase } from './extendDatabase';
+import { PersistenceService } from '../persistence/persistence.service';
+import { ArlasConfigService, ArlasStartupService } from '../startup/startup.service';
+import { ExtendLocalDatabase } from './extendLocalDatabase';
+import { ExtendPersistenceDatabase } from './extendPersistenceDatabase';
 import { Extend } from './model';
-import { ArlasStartupService } from '../startup/startup.service';
+import { HttpClient } from '@angular/common/http';
 
-@Injectable({
-  providedIn: 'root'
-})
+@Injectable()
 export class ArlasExtendService {
-  public dataBase: ExtendDatabase;
+  public dataBase: ExtendLocalDatabase | ExtendPersistenceDatabase;
```

Comment:
Is there no parent class or interface to both ExtendLocalDatabase & ExtendPersistenceDatabase?

Reply:
There is no common thing that could be used here.